xbmc/listproviders/DirectoryProvider.cpp

```diff
@@ -272,7 +274,16 @@ void CDirectoryProvider::OnEvent(const ADDON::AddonEvent& event)
         typeid(event) == typeid(ADDON::AddonEvents::Disabled) ||
         typeid(event) == typeid(ADDON::AddonEvents::InstalledChanged) ||
         typeid(event) == typeid(ADDON::AddonEvents::MetadataChanged))
-    m_updateState = INVALIDATED;
+      m_updateState = INVALIDATED;
+  }
+}
+
+void CDirectoryProvider::Notify(const Observable &obs, const ObservableMessage msg)
+{
+  if (URIUtils::IsProtocol(m_currentUrl, "pvr"))
+  {
+    if (msg == ObservableMessageManagerStateChanged && g_PVRManager.IsStarted())
```

Comment:
Don't call g_PVRManager.IsStarted here please. The directory provider should not need to know about things like this; instead pvr should sent appropriate events.